dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -365,4 +447,35 @@ public String toString() {
             return "EmptyBodyDeserializer{}";
         }
     }
+
+    @SuppressWarnings("unchecked")
+    @Nullable
+    private static <T> T createSuccessType(TypeMarker<T> successT, InputStream inputStream) {
+        if (successT.getType() instanceof Class<?>) {
+            try {
+                return (T) ((Class<?>) successT.getType())
+                        .getConstructor(InputStream.class)
+                        .newInstance(inputStream);
+            } catch (ReflectiveOperationException e) {
+                throw new SafeRuntimeException("Failed to create success type", e);
+            }
+        }
+        return null;
```

Comment:
in the else case, we should throw rather than returning a null ref that will cause something else to throw in a more confusing way later on!

I might suggest splitting this up a bit as well: We should be able to do the validation on `successT.getType()` once when the deserializer is first created, as well as the `.getConstructor(<arg>)`, that way we only need to execute `newInstance` on a per-response basis.